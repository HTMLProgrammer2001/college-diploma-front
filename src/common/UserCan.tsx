import React from 'react';
import {connect, ConnectedProps} from 'react-redux';

import {RootState} from '../redux';
import {selectMeInfo} from '../redux/me/selectors';


const mapStateToProps = (state: RootState) => ({
	user: selectMeInfo(state)
});

const connected = connect(mapStateToProps, null);

type IUserCanProps = {role: number, children: any} & ConnectedProps<typeof connected>;
export const _UserCan: React.FC<IUserCanProps> = ({role, children, user}) => {
	return user.role <= role ? <>{children}</> : null;
};

export default connected(_UserCan);
