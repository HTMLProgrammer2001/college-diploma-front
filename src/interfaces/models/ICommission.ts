export type ICommission = {
	id: number,
	name: string
};
