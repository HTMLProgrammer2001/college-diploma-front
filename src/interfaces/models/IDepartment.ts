export type IDepartment = {
	id: number,
	name: string
};
