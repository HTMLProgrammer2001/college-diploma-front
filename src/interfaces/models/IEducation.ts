import {IUserShort} from './IUserShort';


export type IEducation = {
	id: number,
	institution: string,
	graduate_year: number,
	qualification: number,
	user: IUserShort,
	specialty: string
};
