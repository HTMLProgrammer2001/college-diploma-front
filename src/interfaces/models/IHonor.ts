import {IUserShort} from './IUserShort';


export type IHonor = {
	id: number,
	order: string,
	datePresentation: string,
	title: string,
	active: boolean,
	user: IUserShort,
	type: string
};
