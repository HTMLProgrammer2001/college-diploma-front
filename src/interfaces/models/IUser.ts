export type IUser = {
	id: number,
	fullName: string,
	birthday?: string,
	email: string,
	address?: string,
	role: number,
	avatar?: string,
	experience: number,
	pedagogicalTitle: string,
	category?: string,
	commission: string,
	department: string,
	rank?: string
};
