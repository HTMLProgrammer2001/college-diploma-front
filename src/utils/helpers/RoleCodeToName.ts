import i18next from 'i18next';


export enum Roles{
	ADMIN = 1,
	MODERATOR = 10,
	VIEWER = 30,
	USER = 50
}

let roles: any = {
	[Roles.ADMIN]: 'common.roles.admin',
	[Roles.MODERATOR]: 'common.roles.moderator',
	[Roles.VIEWER]: 'common.roles.viewer',
	[Roles.USER]: 'common.roles.user'
};

const roleCodeToName = (code: number): string => {
	let strCode = code.toString();
	return roles[strCode] ? i18next.t(roles[strCode]) : i18next.t('common.roles.noRole');
};

export default roleCodeToName;
