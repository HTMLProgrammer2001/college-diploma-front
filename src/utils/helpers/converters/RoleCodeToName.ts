import codeConverter from './codeConverter';


export enum Roles{
	ADMIN = 1,
	MODERATOR = 10,
	VIEWER = 30,
	USER = 50
}

let roles: any = {
	[Roles.ADMIN]: 'common.roles.admin',
	[Roles.MODERATOR]: 'common.roles.moderator',
	[Roles.VIEWER]: 'common.roles.viewer',
	[Roles.USER]: 'common.roles.user'
};

export default codeConverter(roles);
