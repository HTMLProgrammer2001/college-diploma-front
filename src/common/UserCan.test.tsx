import React from 'react';
import {shallow} from 'enzyme';

import {_UserCan} from './UserCan';
import {Roles} from '../utils/helpers/converters/RoleCodeToName';
import {IUser} from '../interfaces/models/IUser';


describe('Test user can component unconnected', () => {
	const user: IUser = {id: 1, department: null, commission: null, fullName: '', email: null, role: Roles.USER, experience: 0};

	it('Test render', () => {
		const wrapper = shallow(<_UserCan role={Roles.USER} user={user}>null</_UserCan>);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Test unauthorized', () => {
		const wrapper = shallow(<_UserCan role={Roles.MODERATOR} user={user}>Content</_UserCan>);
		expect(wrapper.html()).toBeFalsy();
	});

	it('Test authorized', () => {
		const content = 'Content',
			wrapper = shallow(<_UserCan role={Roles.USER} user={user}>{content}</_UserCan>);

		expect(wrapper.text()).toBe(content);
	});
});
