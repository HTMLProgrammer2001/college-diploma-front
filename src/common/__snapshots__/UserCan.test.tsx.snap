// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test user can component unconnected Test render 1`] = `"null"`;
